src/Storages/AlterCommands.cpp

```diff
@@ -1245,6 +1245,13 @@ void AlterCommands::prepare(const StorageInMemoryMetadata & metadata)
 {
     auto columns = metadata.columns;
 
+    auto ast_to_str = [](const ASTPtr & query) -> String
+    {
+        if (!query)
+            return "";
+        return queryToString(query);
+    };
```

Comment:
Why not change the `queryToString` implementation to return an empty string if there is no `query` argument?

Reply:
queryToString is a public function that is called in many places. I'm concerned that modifying queryToString might affect a lot of logic.